pom.xml

```diff
@@ -216,6 +222,12 @@
 		</dependency>
 
 		<dependency>
+			<groupId>org.apache.httpcomponents</groupId>
```

Comment:
Why did you add this dependency to HC? It seems that you do not actually depend on it from your code. If it is a transitive dependency from aws-java-sdk, we should not need to depend on it explicitly. I might be missing something, but I'd like to understand...